library/ssl_cli.c

```diff
@@ -2721,7 +2730,7 @@ static int ssl_parse_certificate_request( mbedtls_ssl_context *ssl )
      * therefore the buffer length at this point must be greater than that
      * regardless of the actual code path.
      */
-    if( ssl->in_hslen <= mbedtls_ssl_hs_hdr_len( ssl ) + 3 + n )
+    if( ssl->in_hslen <= mbedtls_ssl_hs_hdr_len( ssl ) + 2 + n )
```

Comment:
I'm not certain, which PR this change belongs to, so I will repeat my comment from another one: I think the comment above must be updated to reflect the new computation.

Reply:
@k-stachowiak Can you clarify how you think the comment should be adapted? At first glance it looks correct to me, but maybe I'm missing something. Both code paths will read `buf[mbedtls_ssl_hs_hdr_len( ssl ) + 2 + n]` so if `ssl->in_hslen <= mbedtls_ssl_hs_hdr_len( ssl ) + 2 + n` we're in trouble and otherwise we're fine (note the operator is `<=` not `<`).  I think that's what the comment says.

Btw, this commit belongs to #1955 - so if you have further comments on it, please make them on that PR's page so they can be addressed. Thanks!